llvm/lib/CodeGen/BranchRelaxation.cpp

```diff
@@ -590,7 +604,7 @@ bool BranchRelaxation::fixupUnconditionalBranch(MachineInstr &MI) {
                             RS.get());
 
   BlockInfo[BranchBB->getNumber()].Size = computeBlockSize(*BranchBB);
-  adjustBlockOffsets(*MBB);
+  adjustBlockOffsets(*MBB, std::next(BranchBB->getIterator()));
 
```

Comment:
My reading of the surrounding code block is that, before fixing up unconditional branch, there is `MBB -> DestBB`, and before this call that it becomes `MBB -> BranchBB -> DestBB -> RestoreBB`, and this line essentially just run `adjustBlockOffsets` for one MBB (the `BranchBB`) even if a range is specified.

I'd suggest add a comment like `create block offsets for newly-created BranchBB` to call out the MBBs adjusted.

Reply:
Added! The path after this function is `MBB -> BranchBB -> RestoreBB -> DestBB`.